modules/claims_api/app/models/claims_api/auto_established_claim.rb

```diff
@@ -384,6 +385,38 @@ def transform_service_branch
       form_data['serviceInformation']
     end
 
+    # Legacy claimsApi code previously allowed servicePay-related service branch
+    # values (servicePay.militaryRetiredPay.payment.serviceBranch, servicePay.separationPay.payment.serviceBranch)
+    # that are not accepted by EVSS.
+    # Rather than refuse those invalid values, this maps them to an equivalent value that EVSS will accept.
+    # Note transform_service_branch above only handles cases found in 'serviceInformation.servicePeriod'.
+    def transform_service_pay_service_branch # rubocop:disable Metrics/MethodLength
+      if form_data['servicePay'].present?
+        if form_data['servicePay']['militaryRetiredPay'].present?
```

Comment:
instead of double nesting you could do a dig and see if the value is present. 

https://apidock.com/ruby/Hash/dig

dig will return nil if at any step there is a nil value.